kong/clustering/data_plane.lua

```diff
@@ -299,17 +278,8 @@ function _M:communicate(premature)
     ngx_log(ngx_ERR, _log_prefix, err_msg, log_suffix)
   end
 
-  -- the config thread might be holding a lock if it's in the middle of an
-  -- update, so we need to give it a chance to terminate gracefully
-  config_exit = true
-
-  ok, err, perr = ngx.thread.wait(config_thread)
-  if not ok then
-    ngx_log(ngx_ERR, _log_prefix, err, log_suffix)
-
-  elseif perr then
-    ngx_log(ngx_ERR, _log_prefix, perr, log_suffix)
-  end
+  -- unregister the recv event handler
+  events.clustering_recv_config(handler, false)
```

Comment:
Is necessary? It seems that the `handler` does not hold closure values and should not need to be registered/deregistered every time connection breaks.

Reply:
No, the handler is a closure, which should catch the local var `ping_immediately` and `next_data` in each communication.

I think I could refactor the code in another PR to eliminate the localization.